### ISSUE TITLE
Documentation Clarification Needed for Clamping of Scale Coefficient in clip_grads_with_norm_

### ISSUE THREAD

**Issue by qiaoqiaoLF** [pytorch/pytorch#151554]
### 📚 The doc issue

In the current documentation for `torch.nn.utils.clip_grads_with_norm_`, the formula for the scale coefficient is as follows:

$$
\text{grad} = \text{grad} \times \frac{\text{maxnorm}}{\text{totalnorm} + 1e-6}
$$

However, in practical usage, it appears that the scale coefficient is clamped to a maximum value of 1, which prevents the gradient from scaling the gradient up. This behavior, while important for the correct functionality of gradient clipping, is not currently mentioned in the documentation.

**Suggested Change:**
It would be helpful to explicitly mention in the documentation that the scale coefficient is clamped to 1. This will provide more clarity to users about how the function operates in practice and ensure there are no misunderstandings regarding its behavior.

### Suggest a potential alternative/fix

The formula should be updated as follows:

$$
\text{grad} = \text{grad} \times min( \frac{\text{maxnorm}}{\text{totalnorm} + 1e-6}, 1)
$$

**Comment by RajeshvShiyal** [comment]
Hello @qiaoqiaoLF, @albanD, @jbschlosser,

If this issue is specific to doc issue, then I would be happy to work on this. Thank you.

CC: @spzala

**Comment by RajeshvShiyal** [comment]
Hello @qiaoqiaoLF,

I found below useful information which explains how "torch.nn.utils.clip_grad_norm_" works.
https://stackoverflow.com/questions/76838254/an-example-of-how-pytorch-clip-grad-norm-works/76838445

CC: @spzala

**Comment by RajeshvShiyal** [comment]
**Problem Summary:**
The current documentation shows the gradient scaling formula as: 
grad = grad * (max_norm / (total_norm + 1e-6))

However, the actual implementation clamps the scale coefficient to a maximum of 1, meaning gradients are never scaled up—only down. This is a crucial detail for users relying on precise gradient behavior.

**Proposed Fix:**
1. Update the Formula in the Docstring
Replace the current formula with:
grad = grad * min(max_norm / (total_norm + eps), 1.0)

2. Add an Explanation Note
Note: The scale coefficient is clamped to a maximum of 1.0 to prevent gradient amplification. This ensures that gradients are only scaled down when the total norm exceeds max_norm.

CC: @spzala, @qiaoqiaoLF
